### PR TITLE
block on forkchoiceUpdated EL calls due to doing fewer of them

### PR DESCRIPTION
**Pull request by tersec**
In a steady-state synced situation, the CL needs to wait anyway for the EL to finish, so it was not truly asynchronous except during syncing.

### CODE REVIEW

File: beacon_chain/gossip_processing/block_processor.nim [hunk 1]
```diff
@@ -543,7 +549,7 @@ proc storeBlock*(
       # - "Beacon chain gapped" from DAG head to optimistic head,
       # - followed by "Beacon chain reorged" from optimistic head back to DAG.
       self.consensusManager[].updateHead(newHead.get.blck)
-      asyncSpawn eth1Monitor.runForkchoiceUpdatedDiscardResult(
+      await eth1Monitor.runForkchoiceUpdatedDiscardResult(
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/block_processor.nim:552]: refactor to put the discard here?

File: beacon_chain/gossip_processing/block_processor.nim [hunk 2]
```diff
@@ -572,10 +578,10 @@ proc storeBlock*(
           # Some attached validator is next proposer, so prepare payload. As
           # updateHead() updated the DAG head, runProposalForkchoiceUpdated,
           # which needs the state corresponding to that head block, can run.
-          asyncSpawn self.consensusManager.runProposalForkchoiceUpdated(
+          await self.consensusManager.runProposalForkchoiceUpdated(
```
Comment (arnetheduck) [inline beacon_chain/gossip_processing/block_processor.nim:581]: it's kind of weird that we sometimes run fcu through consensusmanager and sometimes through eth1monitor - can't we collect all FCU calls in consensus manager? also, can't we move the proposer-or-not logic there?

**Review by arnetheduck — APPROVED** [review]
soo... this PR looks like it might solve the immediate problem of concurrent / stale head updates, but it also feels like the code could be simplified further with a round of refactoring (which we can leave for another PR) - ie it's starting to get quite hairy, to understand what sets which head where and why

**Comment by arnetheduck** [comment]
merging for now since sync is completely broken by the concurrent head updates when spawned tasks get reordered

File: beacon_chain/gossip_processing/block_processor.nim (hunk 2 above)
Comment (tersec) [reply beacon_chain/gossip_processing/block_processor.nim:581]: https://github.com/status-im/nimbus-eth2/pull/4465 already does this in `elManager.forkchoiceUpdated`

**Comment by tersec** [comment]
> soo... this PR looks like it might solve the immediate problem of concurrent / stale head updates, but it also feels like the code could be simplified further with a round of refactoring (which we can leave for another PR) - ie it's starting to get quite hairy, to understand what sets which head where and why

There is potentially other cleanup from cases to consolidate -- previously, one distinction was exactly whether to `asyncSpawn` or `await`. If it always `awaits`, there is less reason to delineate cases so finely.

File: beacon_chain/gossip_processing/block_processor.nim (hunk 1 above)
Comment (tersec) [reply beacon_chain/gossip_processing/block_processor.nim:552]: `runForkchoiceUpdatedDiscardResult` is one such example of a now-too-fine delineation, yes which was there only to support some cases being `asyncSpawn`, indeed, for example.

Comment (tersec) [reply beacon_chain/gossip_processing/block_processor.nim:552]: https://github.com/status-im/nimbus-eth2/pull/4614

**Pull request merged**